api/migrations/0003_profile.py

```diff
@@ -0,0 +1,53 @@
+# Generated by Django 3.1.4 on 2021-02-08 02:10
+
+import django.db.models.deletion
+from django.conf import settings
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ("api", "0002_matchstatus"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="Profile",
+            fields=[
+                (
+                    "id",
+                    models.AutoField(
+                        auto_created=True,
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                ("first_name", models.CharField(blank=True, max_length=50)),
+                ("last_name", models.CharField(blank=True, max_length=50)),
```

Comment:
We will need to max length values to be retrievable by the frontend (unless we decide to hardcode it). Can we pull them out into a config?

Reply:
This is a good point. This is kind of awkward to do because we aren't building the form on the backend e.g. a template which would automatically grab this information from the model.

I'm not really sure of an elegant way to handle this atm with our current stack.

Reply:
Pulling it out into a data file that can be fetched when the client navigates to the login page might be workable.

Reply:
That sounds like what we want. Looking at the gender options, this could be a recurring issue in other areas as well

Reply:
I think the duplication of the max lengths is not too egregious

Reply:
Having to edit the same data in two separate places is bound to cause consistency issues between the frontend and backend. And like eddie mentions, this goes beyond just these two fields. Any constraints on user-provided data on the backend should be reflected in the frontend.

Reply:
If we want frontend form validation, we need to coordinate this somehow. If we choose not to have frontend validation (at least not in the form of input element attributes), then I'd say it's safe to ignore this issue and let the backend say what is invalid (ofc we wouldn't want to be too specific with our error responses for logging in).

Reply:
If we do decide to separate these conditions, then we could make use of it in models.py as well

Reply:
See discord